File: egghead.io/data-structures-and-algorithms-in-js/lesson-14-recursively-sort-an-arr-in-js-with-quick-sort/quick_sort.py
def quick_sort(array):
  if len(array) < 2: return array

  pivot_index = len(array) - 1
  pivot = array[pivot_index]
  pred, succ = [], []

  for i in range(pivot_index):
    curr = array[i]
    if curr < pivot: pred.append(curr)
    else: succ.append(curr)

  sorted_array = (
    quick_sort(pred) 
    + [pivot] 
    + quick_sort(succ)
  )
  print(sorted_array)

  return sorted_array

File: egghead.io/data-structures-and-algorithms-in-js/lesson-14-recursively-sort-an-arr-in-js-with-quick-sort/quick-sort.js
const quickSort = (array) => {
  if (array.length < 2) return array;

  const pivotIndex = array.length - 1;
  const pivot = array[pivotIndex];
  const pred = [];
  const succ = [];

  for (let i = 0; i < pivotIndex; i++) {
    const curr = array[i];
    curr < pivot
      ? pred.push(curr)
      : succ.push(curr);
  }

  const sorted = [
    ...quickSort(pred), 
    pivot, 
    ...quickSort(succ)
  ];
  console.log(sorted);

  return sorted;
};

module.exports = quickSort;
